release/v1alpha1/types.go

```diff
@@ -0,0 +1,171 @@
+	// This condition is terminal
+	//
+	// if Rejected is true the ReleasePayload has failed one or more of its verification criteria
+	// The release-controller will take no more action in this phase, but a human may set the
+	// phase back to Ready to retry and the controller will attempt verification again.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// BlockingJobResults stores the results of all blocking jobs
+	BlockingJobResults []JobStatus `json:"blockingJobResults,omitempty"`
+
+	// InformingJobResults stores the results of all informing jobs
+	InformingJobResults []JobStatus `json:"informingJobResults,omitempty"`
+
+	// AnalysisJobResults stores the results of all analysis jobs
+	AnalysisJobResults []JobStatus `json:"analysisJobResults,omitempty"`
+}
+
+// JobState the aggregate state of the job
+// Supported values include Pending, Failed, and Success.
+type JobState string
+
+const (
+	// JobStatePending not all job runs have completed
+	// Transitions to Failed or Success
+	JobStatePending JobState = "Pending"
+
+	// JobStateFailed failed job aggregation
+	JobStateFailed JobState = "Failed"
+
+	// JobStateSuccess successful job aggregation
+	JobStateSuccess JobState = "Success"
+)
+
+// JobStatus encapsulates the name of the job, all the results of the jobs, and an aggregated
+// result of all the jobs
+type JobStatus struct {
+	// jobName is the name of the job
+	JobName string `json:"name,omitempty"`
+
+	// MaxRetries maximum times to retry a job
+	MaxRetries int `json:"maxRetries,omitempty"`
+
+	// AggregateState is the overall success/failure of all the executed jobs
+	AggregateState JobState `json:"state,omitempty"`
+
+	// JobRunResults contains the links for individual jobs
+	JobRunResults []JobRunResult `json:"results,omitempty"`
+}
+
+// JobRunState the status of a job
+type JobRunState string
+
+const (
+	// JobRunStateTriggered job has been created but not scheduled
+	JobRunStateTriggered JobRunState = "Triggered"
+
+	// JobRunStatePending job is running and awaiting completion
+	JobRunStatePending JobRunState = "Pending"
+
+	// JobRunStateFailure job completed with errors
+	JobRunStateFailure JobRunState = "Failure"
+
+	// JobRunStateSuccess job completed without errors
+	JobRunStateSuccess JobRunState = "Success"
+
+	// JobRunStateAborted job was terminated early
+	JobRunStateAborted JobRunState = "Aborted"
+
+	// JobRunStateError job could not be scheduled
+	JobRunStateError JobRunState = "Error"
+)
+
+// JobRunResult the results of a job run
+// The release-controller creates prowjobs during the sync_ready control loop and relies on an informer to process jobs,
+// that it created, as they are completed. The JobRunResults will be created, by the release-controller during the sync_ready
+// loop and updated whenever any changes, to the respective job is received by the informer.
+type JobRunResult struct {
+	// Name unique name for the job run
+	Name string `json:"name,omitempty"`
```

Comment:
group the prowjob coordinates under a `ProwJobLocation` struct

Reply:
Should this also not be an ObjectReference to the ProwJob?

Reply:
> Should this also not be an ObjectReference to the ProwJob?

We don't use ObjectReferences since the help never matches and the fields are rarely correct.  See the comment upstream about not using them for any new apis.